hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/snapshot/SnapshotUtils.java

```diff
@@ -86,6 +88,13 @@ public static SnapshotInfo getSnapshotInfo(final OzoneManager ozoneManager,
     return snapshotInfo;
   }
 
+  public static SnapshotInfo getSnapshotInfo(OzoneManager ozoneManager,
+                                             SnapshotChainManager chainManager,
+                                             UUID snapshotId) throws IOException {
+    String tableKey = chainManager.getTableKey(snapshotId);
```

Comment:
If the `tableKey` becomes `null` because of some race condition, Then we should handle it gracefully.

Reply:
We can do this as part of another jira. As of now this can only happen because of cache corrupting the snapshot chain.

Reply:
This can be done as part of this jira https://issues.apache.org/jira/browse/HDDS-11452

Reply:
The OM should crash to prevent further state modification, since the OM is already in an IllegalState

Reply:
yup chain corruption would only lead to such inconsistent results